pootle/checks.py

```diff
@@ -262,6 +262,73 @@ def check_settings(app_configs=None, **kwargs):
+            if 'INDEX_NAME' not in settings.POOTLE_TM_SERVER[server]:
+                errors.append(checks.Critical(
+                    _("POOTLE_TM_SERVER['%s'] has no INDEX_NAME.", server),
+                    hint=_("Set an INDEX_NAME for POOTLE_TM_SERVER['%s'].",
+                           server),
+                    id="pootle.C008",
+                ))
+            elif settings.POOTLE_TM_SERVER[server]['INDEX_NAME'] in tm_indexes:
+                errors.append(checks.Critical(
+                    _("Duplicate '%s' INDEX_NAME in POOTLE_TM_SERVER.",
+                      settings.POOTLE_TM_SERVER[server]['INDEX_NAME']),
+                    hint=_("Set different INDEX_NAME for all servers in "
+                           "POOTLE_TM_SERVER."),
+                    id="pootle.C009",
+                ))
+            else:
+                tm_indexes.append(settings.POOTLE_TM_SERVER[server]['INDEX_NAME'])
+
+            if 'WEIGHT' not in settings.POOTLE_TM_SERVER[server]:
+                errors.append(checks.Warning(
+                    _("POOTLE_TM_SERVER['%s'] has no WEIGHT.", server),
+                    hint=_("Set a WEIGHT for POOTLE_TM_SERVER['%s'].", server),
+                    id="pootle.W018",
+                ))
+            elif (settings.POOTLE_TM_SERVER[server]['WEIGHT'] < 0 or
+                  settings.POOTLE_TM_SERVER[server]['WEIGHT'] > 1):
+                errors.append(checks.Critical(
+                    _("POOTLE_TM_SERVER['%s'] doesn't have a WEIGHT less than "
+                      "0 or greater than 1.", server),
+                    hint=_("Set a WEIGHT between 0 and 1 (both included) for "
+                           "POOTLE_TM_SERVER['%s'].", server),
+                    id="pootle.C010",
+                ))
+
+            if 'IS_LOCAL' not in settings.POOTLE_TM_SERVER[server]:
+                errors.append(checks.Critical(
+                    _("POOTLE_TM_SERVER['%s'] has no IS_LOCAL.", server),
+                    hint=_("Set IS_LOCAL for POOTLE_TM_SERVER['%s'].", server),
+                    id="pootle.C011",
+                ))
+            elif not isinstance(settings.POOTLE_TM_SERVER[server]['IS_LOCAL'], bool):
+                errors.append(checks.Warning(
+                    _("POOTLE_TM_SERVER['%s']['IS_LOCAL'] doesn't have a "
+                      "boolean value.", server),
+                    hint=_("Set a boolean value for "
+                           "POOTLE_TM_SERVER['%s']['IS_LOCAL'].", server),
+                    id="pootle.W019",
+                ))
+
+            if 'TM_NAME' not in settings.POOTLE_TM_SERVER[server]:
+                errors.append(checks.Critical(
+                    _("POOTLE_TM_SERVER['%s'] has no TM_NAME.", server),
+                    hint=_("Set TM_NAME for POOTLE_TM_SERVER['%s'].", server),
+                    id="pootle.C012",
+                ))
+            elif not settings.POOTLE_TM_SERVER[server]['TM_NAME']:
```

Comment:
They are not. The first checks if the server settings have a 'TM_NAME' key while the second checks that its value is actually something.

But I am dropping `TM_NAME` from settings, so nevermind.